### DISCUSSION TITLE
Getting Started with Dogecoin Development - Need Guidance, Documentation, and Testnet Information

### DISCUSSION THREAD

**Discussion by yangyile1990** [dogecoin/dogecoin#3485] [Help needed]
I'm a Golang software developer, and my supervisor has assigned me a task to work with Dogecoin. However, I have little to no experience with it, and I feel lost. I'm seeking guidance on how to get started with Dogecoin development.

Here are a few questions and areas where I need assistance:

1. Golang Packages: Could you please recommend a reliable Golang package on GitHub that I can use for Dogecoin development?

2. Connecting to a Node: I would appreciate it if you could suggest a free-to-use Dogecoin node that I can connect to. Additionally, please provide instructions on how to establish a connection with the node.

3. Basic Demo: It would be extremely helpful if you could provide a simple demo or point me to a beginner-friendly tutorial that demonstrates basic Dogecoin functionality using Golang.

4. Documentation: Is there any official or comprehensive documentation available that covers Dogecoin development with Golang? I would greatly benefit from a resource that explains the various APIs, functionalities, and best practices.

5. Testnet: Does Dogecoin have a testnet (test network) available for development and testing purposes? If yes, please provide instructions on how to connect to the Dogecoin testnet and any additional details I should know.

I'm eager to learn and contribute effectively to the Dogecoin project. Any guidance, documentation, or resources you can provide would be greatly appreciated.

Thank you in advance for your help!

Best regards.

**Comment by yangyile1990** [comment]
My code can not work. I don't know the reason.
```
package main

import (
	"github.com/btcsuite/btcd/rpcclient"
)

func main() {
	cfg := &rpcclient.ConnConfig{
		Host:         "rpc.dogecoin.com", // 第三方 DOGE币节点的RPC地址
		User:         "",                 // 留空或设置为一个空字符串
		Pass:         "",                 // 留空或设置为一个空字符串
		DisableTLS:   true,               // DOGE币节点默认不使用TLS
		HTTPPostMode: true,               // DOGE币节点只支持HTTP POST模式
	}
	client, err := rpcclient.New(cfg, nil)
	if err != nil{
		panic(err)
	}

	err = client.Ping()
	if err != nil{
		panic(err)
	}
}
```

**Comment by georgeartem** [comment]
Appears related to discussion #3471 a lot in discussions that could be leveraged more
